packages/client/hmi-client/src/page/project/components/tera-upload-resources-modal.vue

```diff
@@ -221,29 +221,29 @@ function importCompleted(newResults: { id: string; name: string; assetType: Asse
 
 async function upload() {
 	if (results.value) {
-		await Promise.all(
-			results.value?.map(({ id, assetType, name }): Promise<any> => {
-				const newAsset = useProjects().addAsset(assetType, id);
-				if (name && name.toLowerCase().endsWith('.pdf')) {
-					extractPDF(id);
-				} else if (
-					(name && name.toLowerCase().endsWith('.txt')) ||
-					(name && name.toLowerCase().endsWith('.md'))
-				) {
-					modelCard(id);
-				}
-				return newAsset;
-			})
-		).then((resolved) => {
-			emit('close');
-			const resourceMsg = resolved.length > 1 ? 'resources were' : 'resource was';
-			useToastService().success(
-				'Success!',
-				`${resolved.length} ${resourceMsg} successfuly added to this project`
-			);
-			importedFiles.value = [];
-			results.value = null;
+		const createAssetsPromises = (results.value ?? []).map(({ id, assetType }) =>
+			useProjects().addAsset(assetType, id)
+		);
+		const createdAssets = await Promise.all(createAssetsPromises);
+		createdAssets.forEach((_, index) => {
+			const { name, id } = (results.value ?? [])[index];
+			if (name && name.toLowerCase().endsWith('.pdf')) {
+				extractPDF(id);
+			} else if (
+				name &&
+				(name.toLowerCase().endsWith('.txt') || name.toLowerCase().endsWith('.md'))
+			) {
+				modelCard(id);
+			}
```

Comment:
I'm still wondering if those calls should be done on the backend instead. I don't see why the front-end should worry about those 🤔

Reply:
Yeah I think that makes sense. Since the extraction is part of creating an assets in the project. It makes sense to handle those additional operation on the backend especially If it's always the case we run those procesess after creating an assets.